fields/S1G.md

```diff
@@ -0,0 +1,48 @@
+---
+title: S1G
+categories: [suggested]
+---
+Type
+: 32 
+
+Structure
+: u16 known, u16 data1, u16 data2,
+
+Required Alignment
+: 2
+
+Unit(s)
+: none
+
+The presence of this field indicates the frame was capture using an S1G phy.
+
+## known
+
+This field indicates the (known) contents of the S1G.
+
+| **`0x0001`** | S1G PPDU Format Known |
+| **`0x0002`** | Response Indication Known |
+| **`0x0004`** | Guard Interval Known |
+| **`0x0008`** | NSS Known | 
+| **`0x0010`** | Bandwidth Known |
+| **`0x0020`** | MCS Known |
+| **`0x0040`** | Color Known |
+| **`0x0080`** | Uplink Indication Known |
+| **`0xFF00`** | Reserved |
+
+## data1
+
+| **`0x0003`** | S1G PPDU Format: 0=S1G_1M, 1=S1G_SHORT, 2=S1G_LONG |
+| **`0x000C`** | Response Indication: 0=NO-RESPONSE, 1=NDP_RESPONSE, 2=NORMAL_RESPONSE, 3=LONG_RESPPNSE |
+| **`0x0010`** | Reserved |
+| **`0x0020`** | Guard Interval: 0=Long GI, 1=Short GI |
+| **`0x00C0`** | Number Spatial Streams: 0=1 Spatial stream, 1=2, .. 3=4 Spatial streams |
+| **`0x0F00`** | Bandwidth: 0=1MHz, 1=2MHz, 2=4MHz, 3=8MHz, 4=16MHz, 5-15 reserved |
+| **`0xF000`** | MCS (MCS rate index, 0-10, 11-15 reserved) |
+
+## data2
+
+| **`0x0007`** | Color: 0-7 |
+| **`0x0008`** | Uplink indication |
+| **`0x00F0`** | Reserved |
+| **`0xFF00`** | RSSI |
```

Comment:
there's no 'known' bit for the RSSI, but it shouldn't be needed anyway? there should be a separate "Antenna signal" for it, no?

Reply:
Yes, there is no "known" bit for RSSI. These definition is finalized and test during program announcement. I just post the definition based on current imeplementation.

Reply:
I understand that, but that doesn't mean it's good?

This is what you get for not following the right process I guess.

Personally, I think the official version shouldn't even carry this field.

Also, come to think of it, the process says there should be a producer we can look at ...

I suppose it's better to have this discussion on the mailing list.

Reply:
Each frame during driver capture should always have RSSI value. So we should not need "known" bit

Reply:
That's maybe true for your implementation now, but it shouldn't be _required_ of an implementation?

Reply:
Let me check the sample sniffer and confirm if there is "Antenna signal" field.

Reply:
That kind of seems the wrong way around ... this discussion should've happened years ago, and we *did* have this discussion (with Richard), but nobody followed up on your side.

Reply:
There actually is a possible difference between signal and RSSI, but here you don't even really define what RSSI means, and even in that case it should probably not be S1G specific, but another field.

Reply:
I check the proposal again in https://lore.kernel.org/radiotap/ and our implementation. I also confirmed the radiotap headers driver provided in S1G sniffer capture. I will only keep the consistent definition.

Reply:
Do I need to remove Uplink indication? the field is not in the proposal.

Reply:
I also removed uplink indication to sync with the original proposal.
Please help to check if there is any problem.